common/selector/weighted_selector.go

```diff
@@ -0,0 +1,130 @@
+package selector
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"slices"
+
+	"github.com/Layr-Labs/eigensdk-go/logging"
+)
+
+// Performs weighted random selection with configurable filtering of low performers.
```

Comment:
If this is generic, "low performers" may not make sense. It's conceivable we'd want to use this utility in a scenario where weights don't correspond to performance.